api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -224,6 +227,17 @@ public void purgeObs(Obs obs, boolean cascade) throws APIException {
 	}
 	
 	/**
+	 * @see org.openmrs.api.ObsService#purgeObs(org.openmrs.Obs, boolean)
+	 */
+	public void purgeObs(Obs obs, boolean cascade) throws APIException {
+		if (purgeComplexData(obs) == false) {
```

Comment:
The possible implementations for complex obs handlers are too broad to just reduce the behaviour of `purgeComplexData(..)` like that, IMHO. In reality this method should return some form of success or status code that can be unequivocally linked to a clearly identified situation.

I will settle on @djazayeri suggestion to add a `forcePurge` parameter that permits ignoring non-DAO deletions. It will of course default to false.